app/src/main/java/org/fossasia/openevent/app/core/event/create/CreateEventFragment.java

```diff
@@ -82,50 +85,36 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
-            Bundle bundle = ai.metaData;
-            String placesApiKey = bundle.getString("com.google.android.geo.API_KEY");
-            if ("YOUR_API_KEY".equals(placesApiKey)) {
-                Timber.d("Add Google Places API key in AndroidManifest.xml file to use Place Picker.");
-                binding.form.buttonPlacePicker.setVisibility(View.GONE);
-                binding.form.layoutLatitude.setVisibility(View.VISIBLE);
-                binding.form.layoutLongitude.setVisibility(View.VISIBLE);
-                showLocationLayouts();
-            }
-        } catch (PackageManager.NameNotFoundException e) {
-            Timber.e(e, "Package name not found");
-        }
+        setupSpinners();
 
-        binding.form.buttonPlacePicker.setOnClickListener(view -> {
-            int errorCode = googleApiAvailabilityInstance.isGooglePlayServicesAvailable(getContext());
-            if (errorCode == ConnectionResult.SUCCESS) {
-                //SUCCESS
-                PlacePicker.IntentBuilder builder = new PlacePicker.IntentBuilder();
-                try {
-                    startActivityForResult(builder.build(getActivity()), PLACE_PICKER_REQUEST);
-                } catch (GooglePlayServicesRepairableException e) {
-                    Timber.d(e, "GooglePlayServicesRepairable");
-                } catch (GooglePlayServicesNotAvailableException e) {
-                    Timber.d("GooglePlayServices NotAvailable => Updating or Unauthentic");
-                }
-            } else if (googleApiAvailabilityInstance.isUserResolvableError(errorCode)) {
-                //SERVICE_MISSING, SERVICE_VERSION_UPDATE_REQUIRED, SERVICE_DISABLED
-                googleApiAvailabilityInstance.getErrorDialog(
-                    getActivity(), errorCode, PLACE_PICKER_REQUEST, (dialog) -> {
-                        showLocationLayouts();
-                    });
-            } else {
-                //SERVICE_UPDATING, SERVICE_INVALID - can't use place picker - must enter manually
-                showLocationLayouts();
+        setupPlacePicker();
+
+        return binding.getRoot();
+    }
+
+    private void setupSpinners() {
+        currencyAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_item);
+        currencyAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        paymentCountryAdapter = new ArrayAdapter<>(getActivity(), android.R.layout.simple_spinner_item);
+        paymentCountryAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        binding.form.paymentCountrySpinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int i, long l) {
+                getPresenter().getEvent().paymentCountry = adapterView.getItemAtPosition(i).toString();
```

Comment:
View should not handle any logic

Reply:
1. I am unable to bind `paymentCountry` in XML.
2. Invoking a `presenter` method using lambda expression in XML seems to not work. 
3. Calling a presenter method from Listener in Fragment will require one more method in Fragment that the Presenter can call, or else Presenter will require accessing Binding variable.
4. A lengthier solution is to use inverse data binding.

Please suggest how to handle this.

Reply:
You shouldn't do it in XML at all. The view should just tell presenter that this event has happened and presenter should handle the rest

Reply:
@iamareebjamal I believe you meant approach no. 3. 
I have made changes. Please review.